tests/pytorch/test_fused_attn.py

```diff
@@ -126,14 +132,142 @@ def _run_dot_product_attention(dtype, bs, config, backend, ckpt_attn, bias_type)
     q = inp[:, :,0,:,:]
     k = inp[:, :,1,:,:]
     v = inp[:, :,2,:,:]
-    op = block(q, k, v, attn_mask_type=config.attn_mask_type,
+    op = block(q, k, v,
+        qkv_format='sbhd',
+        cu_seqlens_q = cu_seqlens,
+        cu_seqlens_kv = cu_seqlens,
+        attn_mask_type=config.attn_mask_type,
         checkpoint_core_attention=ckpt_attn,
         core_attention_bias_type=bias_type,
         core_attention_bias=bias)
     op.backward(op_grad)
 
     return op, inp.grad
 
+qkv_layouts = [
+    'sb3hd', 'sbh3d', 'sbhd_sb2hd', 'sbhd_sbh2d', 'sbhd_sbhd_sbhd',
+    'bs3hd', 'bsh3d', 'bshd_bs2hd', 'bshd_bsh2d', 'bshd_bshd_bshd',
+    # will add tests for thd layouts later when the support is available in fused attention
+    #'t3hd', 'th3d', 'thd_t2hd', 'thd_th2d', 'thd_thd_thd',
+    ]
+
+@pytest.mark.skipif(
+    get_device_compute_capability() < 8.0, reason="Compute capability 8.0+ is required.")
+@pytest.mark.parametrize("dtype", param_types)
+@pytest.mark.parametrize("bs", batch_sizes)
+@pytest.mark.parametrize("model", model_configs.keys())
+@pytest.mark.parametrize("workspace_opt", [True, False])
+@pytest.mark.parametrize("qkv_layout", qkv_layouts)
+def test_dpa_qkv_layout(dtype, bs, model, workspace_opt, qkv_layout):
```

Comment:
It's probably fine for this PR, but I'm starting to worry about how many cases our CI is running. I count 400 cases here, some of which are repeated in `test_dot_product_attention`. In the future, it might be worth refactoring into multiple tests that each only loop over a few variables.

Reply:
It's true that there's a significant increase in CI tests due to the layout expansion. I commented out the bs=32 tests also because they took too long (they take much longer than the smaller bs ones), and our H100 CI timed out.

All the tests were added for a reason - for example, we had issues with bs=32 and some of the layouts before - but I do think we need to cut down the number of tests especially on the attention side. I'll look into it in the next PR.